compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp

```diff
@@ -300,9 +314,29 @@ namespace cppmicroservices {
       return GetConfiguration(pid);
     }
 
-    std::vector<std::shared_ptr<cppmicroservices::service::cm::Configuration>> ConfigurationAdminImpl::ListConfigurations(const std::string& /* filter */)
+    std::vector<std::shared_ptr<cppmicroservices::service::cm::Configuration>> ConfigurationAdminImpl::ListConfigurations(const std::string& filter)
     {
-      throw std::invalid_argument("Method not currently implemented");
+      std::vector<std::shared_ptr<cppmicroservices::service::cm::Configuration>> result;
+      {
+        std::lock_guard<std::mutex> lk {configurationsMutex};
+
+        if (filter.empty()) {
+          result.reserve(configurations.size());
+          for (auto it : configurations) {
```

Comment:
Is this right? if "configurations" is a vector, what's used by the range-based for isn't an iterator.